clusterctl/cmd/validate.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+var validateCmd = &cobra.Command{
+	Use:   "validate",
+	Short: "Validate an API resource created by cluster API.",
+	Long:  "Validate an API resource created by cluster API.",
+}
+
+func init() {
+	validateCmd.AddCommand(validateClusterCmd)
```

Comment:
To reduce confusion, we should be consistent with who is responsible for adding a sub-command. Since the validate command adds itself to the RootCmd on line 31, then it seems like we should remove line 30 and have the ValidateClusterCmd add itself to the ValidateCmd.

Reply:
By the way, is there a repo wide style guideline?

Reply:
As you pointed out there's inconsistency in the repo. We can fix that in a follow up PR. Let's follow the style laid out in the cobra examples [1] where a given command adds itself to its parent. So that means the create command should be added in create.go. And in this case, the validateClusterCommand would add itself to the validate command in validate_cluster.go.

[1] https://github.com/spf13/cobra

Reply:
@spew - thanks for standardizing
+1 for following the standardized pattern.

Reply:
Done